hummingbot/smart_components/executors/position_executor/position_executor.py

```diff
@@ -381,7 +381,7 @@ def process_order_failed_event(self, _, market, event: MarketOrderFailureEvent):
         elif self.close_order.order_id == event.order_id:
             self.place_close_order(self.close_type)
         elif self.take_profit_order.order_id == event.order_id:
-            self.place_take_profit_limit_order()
+            self.take_profit_order.order_id = None
```

Comment:
Could you please add more info, why you chose this solution?
Here `place_take_profit_limit_order()` will recreate take_profit_order order and assign new id to `self.take_profit_order.order_id`. So it should work as it is now.